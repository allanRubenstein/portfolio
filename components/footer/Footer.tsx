import Image from 'next/image';
import Link from 'next/link';
import React from 'react';
import styled from 'styled-components';
import ScreenReaderOnly from '../common/screen-reader-only/ScreenReaderOnly';
import { ColorsEnum } from '../typography/types';

export interface FooterProps {
  classes?: string;
}

const Footer = ({ classes }: FooterProps) => {
  return (
    <FooterWrap>
      <ImageWrap>
        <Link
          // TODO: add real link
          href="/"
        >
          <a>
            <Image
              src="/images/nav/nav-logo.png"
              alt="logo"
              width={52}
              height={60}
              layout={'responsive'}
            />
            <ScreenReaderOnly>Home Page</ScreenReaderOnly>
          </a>
        </Link>
      </ImageWrap>
    </FooterWrap>
  );
};

export default Footer;

const FooterWrap = styled.footer`
  width: 100%;
  background-color: var(${ColorsEnum.white});
  border-top: 1px solid var(${ColorsEnum.gray});
`;
const ImageWrap = styled.div`
  padding: 3rem 0;
  width: 3rem;
  height: auto;
  display: block;
  margin: auto;
`;
