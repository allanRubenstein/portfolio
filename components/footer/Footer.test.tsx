import React from 'react';
import { render } from '@testing-library/react';
import { axe } from 'jest-axe';
import Footer from './Footer';

test('snapshot', () => {
  const { container } = render(<Footer />);
  expect(container).toMatchSnapshot();
});

test('a11y', async () => {
  const { container } = render(<Footer />);
  const results = await axe(container);
  expect(results).toHaveNoViolations();
});
