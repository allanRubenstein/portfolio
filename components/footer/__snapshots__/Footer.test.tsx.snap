// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c2 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border-width: 0;
}

.c0 {
  width: 100%;
  background-color: var(--white);
  border-top: 1px solid var(--gray);
}

.c1 {
  padding: 3rem 0;
  width: 3rem;
  height: auto;
  display: block;
  margin: auto;
}

<div>
  <footer
    class="c0"
  >
    <div
      class="c1"
    >
      <a
        href="/"
      >
        <div
          style="display: block; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="display: block; box-sizing: border-box; padding-top: 115.38461538461537%;"
          />
          <img
            alt="logo"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <span
          class="c2"
        >
          Home Page
        </span>
      </a>
    </div>
  </footer>
</div>
`;
